ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java

```diff
@@ -47,12 +49,16 @@
     private final String verificationRulesEtag;
     private final ValueSetDataService valueSetDataService;
 
-    public VerificationRulesControllerV2(ValueSetDataService valueSetDataService)
+
+    public VerificationRulesControllerV2(ValueSetDataService valueSetDataService, String[] disabledVerificationModes)
             throws IOException, NoSuchAlgorithmException {
         ObjectMapper mapper = new ObjectMapper();
         InputStream verificationRulesFile =
                 new ClassPathResource("verificationRulesV2.json").getInputStream();
         this.verificationRules = mapper.readValue(verificationRulesFile, Map.class);
+        for(String disabledMode: disabledVerificationModes){
+            ((Map<String, List<Map<String,String>>>)verificationRules.get("modeRules")).get("activeModes").removeIf(mode -> mode.get("id").equals(disabledMode));
+        }
```

Comment:
Maybe it makes sense to use `mapper.readTree` instead to work on `JsonNode` instead of the cast.

Reply:
I agree that the cast is ugly, but won't we just end up casting everything to `ObjectNode` then?  Wouldn't really make much of a difference IMO, we need to cast every field anyway unless we define a model

Reply:
Just tried it, we could save some casting, but `ArrayNode` doesn't have a convenient `remove` method, we'd have to juggle indexes, trading one kind of ugly for another...

Reply:
Ja, removing from arrays is not so cool with JsonNode

Reply:
Ahm I don't think we need any casts as `JsonNode` provides us with the needed functions.
 
Not sure about an easy replacement of `removeIf` though

Reply:
When working with `JsonNode` you would need to replace the provided node, I would not suggest on juggling with indices :D

Reply:
But yeah I guess leave it, rather add a test.